source/unless.js

```diff
@@ -11,7 +11,7 @@ import _curry3 from './internal/_curry3.js';
  * @memberOf R
  * @since v0.18.0
  * @category Logic
- * @sig (a -> Boolean) -> (a -> a) -> a -> a
+ * @sig (a -> Boolean) -> (a -> b) -> a -> a | b
```

Comment:
I'm not too well versed in HM notation so this may be a stupid question:

I think it makes sense to emphasise that `whenFalseFn` and `whenTrueFn` both don't need to return a value of the same type at their input. So changing `a -> a` to `a -> b` (where `b` may or may not be the same type as `a`) makes sense to me.

If `b` can indeed be of the same type as `a` then what's the `a | b` for? I would have expected this instead:

```none
(a -> Boolean) -> (a -> b) -> a -> b
```

What do you think?

Reply:
In `R.when`, If the input value satisfies the predication, `R.when` will execute `whenTrueFn`, and return the result of `whenTrueFn`, otherwise, `R.when` will just return the input value, as follows:

```js
const formatDateIfNotNil = R.unless(isNil, time => moment(time).format('YYYY-MM-DD'));

let time;
formatDateIfNotNil(time); //=> undefined

time = 1635004800000
formatDateIfNotNil(time); //=> '2021-10-24'
```

So if the input value's type is `a`, and the return type of `whenTrueFn` is `b`, `R.when`'s return type will be either `b` or `a`. 

The return type of `R.unless` is the same as `R.when`.

Reply:
Oh yeah I forgot that about both `when` and `unless`. Ok that makes more sense. Thanks